colander/__init__.py

```diff
@@ -2120,6 +2120,8 @@ def clone(self):
         dictionaries are preserved."""
         children = [node.clone() for node in self.children]
         cloned = self.__class__(self.typ, *children)
+        cloned.children = []
+        _add_node_children(cloned, children)
```

Comment:
@tseaver No, the the problem is not in the test, but due to the fact that ``SequenceSchema.__init__`` has to be called with exactly one child node.

Thus attempting to clone a ``SequenceSchema`` threw an exception with the implementation of ``_SchemaNode.clone`` before  b5d4207 / #212.  The current implementation leads to the regression described in this PR.

An alternative to the patch above could be to completely revert b5d4207 and override the ``clone`` method for ``SequenceSchema`` with the code from that change.  I've tested this and all tests pass with this solution.  However, I'm unsure whether this would be a better fix.